secant/Features/BackupFlow/RecoveryPhraseValidation.swift

```diff
@@ -0,0 +1,232 @@
+struct ValidationWord: Equatable {
+    var groupIndex: Int
+    var word: String
+}
+
+struct RecoveryPhraseValidationState: Equatable {
+    enum Step: Equatable {
+        case initial
+        case incomplete
+        case complete
+    }
+
+    enum Route: Equatable, CaseIterable {
+        case success
+        case failure
+    }
+
+    static let wordGroupSize = 6
+    static let phraseChunks = 4
+
+    var phrase: RecoveryPhrase
+    var missingIndices: [Int]
+    var missingWordChips: [PhraseChip.Kind]
+    var validationWords: [ValidationWord]
+    var route: Route?
+    
+    var step: Step {
+        guard !validationWords.isEmpty else {
+            return  .initial
+        }
+
+        guard validationWords.count >= missingIndices.count else {
+            return .incomplete
+        }
+
+        return .complete
+    }
+
+    var isValid: Bool {
+        self.resultingPhrase == phrase.words
+    }
+}
+
+extension RecoveryPhraseValidationViewStore {
+    func bindingForRoute(_ route: RecoveryPhraseValidationState.Route) -> Binding<Bool> {
+        self.binding(
+            get: { $0.route == route },
+            send: { isActive in
+                return .updateRoute(isActive ? route : nil)
+            }
+        )
+    }
+}
+
+extension RecoveryPhraseValidationState {
+    /// creates an initial `RecoveryPhraseValidationState` with no completions and random missing indices.
+    /// - Note: Use this function to create a random validation puzzle for a given phrase.
+    static func random(phrase: RecoveryPhrase) -> Self {
+        let missingIndices = Self.randomIndices()
+        let missingWordChipKind = phrase.words(fromMissingIndices: missingIndices)
+        return RecoveryPhraseValidationState(
+            phrase: phrase,
+            missingIndices: missingIndices,
+            missingWordChips: missingWordChipKind,
+            validationWords: []
+        )
+    }
+}
+
+extension RecoveryPhraseValidationState {
+    /// Given an array of RecoveryPhraseStepCompletion, missing indices, original phrase and the number of groups it was split into,
+    /// assembly the resulting phrase. This comes up with the "proposed solution" for the recovery phrase validation challenge.
+    /// - returns:an array of String containing the recovery phrase words ordered by the original phrase order.
+    var resultingPhrase: [String] {
+        assert(missingIndices.count == validationWords.count)
+        assert(validationWords.count == Self.phraseChunks)
+
+        var words = phrase.words
+        let groupLength = words.count / Self.phraseChunks
+        // iterate based on the completions the user did on the UI
+        for wordCompletion in validationWords {
+            // figure out which phrase group (chunk) this completion belongs to
+            let i = wordCompletion.groupIndex
+            // validate that's the right number
+            assert(i < Self.phraseChunks)
```

Comment:
I'm thinking maybe the comments in the thread above this one aren't coming through because the conversation is resolved.  Just double checking ☝️ .